### PR TITLE
chore: rename AvailabilityKind to Availability [WPB-765]

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-765" title="WPB-765" target="_blank">StoryWPB-765</a>  [iOS] 5.3.16 Indicate verified user
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

As a preparation for PR https://github.com/wireapp/wire-ios/pull/916 this PR renames an enum type.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Source/Model/User/Availability/Availability.swift [hunk 1]
```diff
@@ -16,13 +16,7 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
-import Foundation
-import WireDataModel
-
-extension ZMUser {
-    func updateAvailability(_ newValue: AvailabilityKind) {
```
Comment (caldrian) [inline wire-ios-data-model/Source/Model/User/Availability/Availability.swift:23]: This code was duplicate, see [ZMUser+Availability.swift](https://github.com/wireapp/wire-ios/pull/917/files#diff-8324f473a067274ac5ec7162e57e737adde4bba8358a2a81f2ed28d4c3d463f2R125)

File: wire-ios-data-model/Source/Model/User/Availability/Availability.swift [hunk 2]
```diff
@@ -16,13 +16,7 @@
...
-        self.willChangeValue(forKey: AvailabilityKey)
-        self.setPrimitiveValue(NSNumber(value: newValue.rawValue), forKey: AvailabilityKey)
-        self.didChangeValue(forKey: AvailabilityKey)
-    }
+@objc(ZMAvailability)
+public enum Availability: Int, CaseIterable {
```
Comment (caldrian) [inline wire-ios-data-model/Source/Model/User/Availability/Availability.swift:20]: GitHub shows a relationship between two different files here.

File: wire-ios-data-model/Source/Model/User/Availability/Availability+WireProtos.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            self = .available
+        case .away:
+            self = .away
+        case .busy:
+            self = .busy
+        }
+    }
+}
+
+extension WireProtos.Availability {
```
Comment (theswiftway) [inline wire-ios-data-model/Source/Model/User/Availability/Availability+WireProtos.swift:37]: **(blocking)suggestion:** Aren't these same? Would moving it to a common place and import where ever these are used help in removing this duplication? We do import WireProtos in WireDataModel. I guess we could reuse that. Am I missing something here?

Comment (caldrian) [reply wire-ios-data-model/Source/Model/User/Availability/Availability+WireProtos.swift:37]: `WireProtos.Availability` is generated from Rust code and is a struct, which cannot be used in a Core Data entity. That's why there is another enum which can be used by Obj-C code.
```
@objc(ZMAvailability)
public enum Availability: Int, CaseIterable {
    case none, available, busy, away
}
```

**Review by theswiftway — APPROVED** [review]
LGTM

**Review by agisilaos — APPROVED** [review]

**Pull request merged**